e2e/specs/deep_linking/deeplink.test.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+'use strict';
+const fs = require('fs');
+
+const env = require('../../modules/environment');
+const {asyncSleep} = require('../../modules/utils');
+
+describe('application', function desc() {
+    this.timeout(30000);
+
+    const config = env.demoConfig;
+
+    beforeEach(async () => {
+        env.cleanDataDir();
+        env.createTestUserDataDir();
+        env.cleanTestConfig();
+        fs.writeFileSync(env.configFilePath, JSON.stringify(config));
+        await asyncSleep(1000);
```

Comment:
Was considering that, I was hesitant because we have a lot of tests that control their own configs and app instances. I'll leave it for now but will probably change later. Thanks :)